rest_framework_json_api/views.py

```diff
@@ -98,12 +101,87 @@ def get_queryset(self, *args, **kwargs):
         return qs
 
 
-class ModelViewSet(AutoPrefetchMixin, PrefetchForIncludesHelperMixin, viewsets.ModelViewSet):
+class RelatedMixin(object):
+    """
+    This mixin handles all related entities, whose Serializers are declared in "related_serializers"
+    """
+    related_serializers = {}
+    related_field_mapping = {}
+
+    def retrieve_related(self, request, *args, **kwargs):
+        serializer_kwargs = {}
+        instance = self.get_related_instance()
+
+        if hasattr(instance, 'all'):
+            instance = instance.all()
+
+        if callable(instance):
+            instance = instance()
+
+        if instance is None:
+            return Response(data=None)
+
+        if isinstance(instance, Iterable):
+            serializer_kwargs['many'] = True
+
+        serializer = self.get_serializer(instance, **serializer_kwargs)
+        return Response(serializer.data)
+
+    def get_serializer_class(self):
+        parent_serializer_class = super(RelatedMixin, self).get_serializer_class()
+
+        if 'related_field' in self.kwargs:
+            field_name = self.kwargs['related_field']
+
+            assert hasattr(parent_serializer_class, 'included_serializers')\
+                or self.related_serializers,\
```

Comment:
This looks goo to me. One more though on `related_serializers`.

I think we should move it to the serializer because of following reasons:
1. related fields are defined on the serializer not the view so to define `related_serializers` on serializer feels more natural to me
2. as `included_serializers` is also on serializer it is more consistent. User can then simply decide whether he simply wants related serializers or included serializers at the same spot.
3. Serializers have meta classing. This is out of scope for this PR but in the future we can do `import_class_from_dotted_path` during loading of classes and not on each requests. `PolymorphicModelSerializer` already does meta classing something we should introduce for `ModelSerializer` in the future as well.

What do you think?

Reply:
Actually I do think that related urls and "calculating" related entities is not a part of serializer itself, and should be done in the RelatedMixin. But I'm also ok to do it in the serializer since it just works.